umbral/pre.py

```diff
@@ -381,34 +342,36 @@ def _check_challenge(capsule: Capsule, cfrag: CapsuleFrag,
     xcomp = cfrag.point_eph_ni
     kfrag_id = cfrag.bn_kfrag_id
 
-    e2 = challenge_resp.point_eph_e2
-    v2 = challenge_resp.point_eph_v2
+    e2 = proof.point_eph_e2
+    v2 = proof.point_eph_v2
 
     g = params.g
     u = params.u
 
-    u1 = challenge_resp.point_kfrag_commitment
-    u2 = challenge_resp.point_kfrag_pok
+    u1 = proof.point_kfrag_commitment
+    u2 = proof.point_kfrag_pok
 
-    z1 = challenge_resp.bn_kfrag_sig1
-    z2 = challenge_resp.bn_kfrag_sig2
-    z3 = challenge_resp.bn_sig
+    z1 = proof.bn_kfrag_sig1
+    z2 = proof.bn_kfrag_sig2
+    z3 = proof.bn_sig
 
     g_y = (z2 * g) + (z1 * pub_a)
 
     hash_input = [e, e1, e2, v, v1, v2, u, u1, u2]
-    if challenge_metadata is not None:
-        hash_input.append(challenge_metadata)
+    if proof.metadata is not None:
+        hash_input.append(proof.metadata)
     
     h = BigNum.hash_to_bn(*hash_input, params=params)
 
-    check31 = z1 == BigNum.hash_to_bn(g_y, kfrag_id, pub_a, pub_b, u1, xcomp, params=params)
-    check32 = z3 * e == e2 + (h * e1)
-    check33 = z3 * u == u2 + (h * u1)
-    check34 = z3 * v == v2 + (h * v1)
-
-    return check31 & check32 & check33 & check34
-
+    valid_kfrag_signature     = z1 == BigNum.hash_to_bn(*[g_y, kfrag_id, pub_a, pub_b, u1, xcomp], params=params)
```

Comment:
You don't need to do `*[g_y, kfrag_id, pub_a, pub_b, u1, xcomp]` here.
You can simply do: `BigNum.hash_to_bn(g_y, kfrag_id, pub_a, pub_b, u1, xcomp, params=params)`.

Reply:
Furthermore, not sure if I like doing this oracle operation and the check on the same line. I'll defer to other opinions on this, though.

Reply:
Fixed